File: apps/web/src/lib/speech/speech-error-messages.ts
/** Traduz os codigos de erro da Web Speech API. `null` significa "nao mostre nada ao usuario". */
export function speechErrorMessage(code: string): string | null {
  switch (code) {
    case "aborted":
      return null;
    case "not-allowed":
    case "service-not-allowed":
      return "Permita o acesso ao microfone para gravar.";
    case "no-speech":
      return "Não ouvi nada. Tente de novo.";
    case "audio-capture":
      return "Nenhum microfone encontrado.";
    case "network":
      return "Sem conexão com o serviço de voz.";
    default:
      return "Não foi possível gravar o áudio.";
  }
}

File: apps/mobile/app/src/main/java/app/braid/mobile/ui/voice/SpeechErrorMessages.kt
package app.braid.mobile.ui.voice

import android.speech.SpeechRecognizer

fun speechErrorMessage(
    code: Int,
    stoppedByUser: Boolean = false,
): String? {
    if (stoppedByUser || code == SpeechRecognizer.ERROR_CLIENT) return null
    return when (code) {
        SpeechRecognizer.ERROR_INSUFFICIENT_PERMISSIONS -> "Permita o acesso ao microfone para gravar."
        SpeechRecognizer.ERROR_NO_MATCH,
        SpeechRecognizer.ERROR_SPEECH_TIMEOUT,
        -> "Não ouvi nada. Tente de novo."
        SpeechRecognizer.ERROR_AUDIO -> "Nenhum microfone encontrado."
        SpeechRecognizer.ERROR_NETWORK,
        SpeechRecognizer.ERROR_NETWORK_TIMEOUT,
        SpeechRecognizer.ERROR_SERVER,
        SpeechRecognizer.ERROR_SERVER_DISCONNECTED,
        -> "Sem conexão com o serviço de voz."
        else -> "Não foi possível gravar o áudio."
    }
}
